Gui 1489 improvements to stack detail page and wizard

dkavanagh opened the pull request:
https://eucalyptus.atlassian.net/browse/GUI-1489

euca-qa-gui-tester commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/150/
Test PASSed.

euca-qa-gui-tester commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/173/
Test FAILed.

kamalgill commented:
@dkavanagh: Heads up that https://github.com/eucalyptus/eucaconsole/pull/1331/files#diff-6885d8bf6482cbb40dac62860b22d7c3R130 is missing the closing 'a' tag, which breaks i18n due to the invalid XHTML.

euca-qa-gui-tester commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/175/
Test FAILed.

euca-qa-gui-tester commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/176/
Test FAILed.

euca-qa-gui-tester commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/178/
Test FAILed.

euca-qa-gui-tester commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/179/
Test FAILed.

euca-qa-gui-tester commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/183/
Test FAILed.

euca-qa-gui-tester commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/188/
Test FAILed.

euca-qa-gui-tester commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/189/
Test FAILed.

euca-qa-gui-tester commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/190/
Test PASSed.

kamalgill commented on eucaconsole/static/css/font-awesome/font-awesome.css:
```diff
@@ -65,1043 +62,1043 @@
 
 /* Font Awesome uses the Unicode Private Use Area (PUA) to ensure screen readers do not read off random characters that represent icons */
-.fa-glass:before { content: "\f000"; }
+.fa-glass:before { content: ""; }
```
@dkavanagh: Not sure if this is a display issue on GitHub, but the Font Awesome content references all seem broken

dkavanagh replied:
It's github. That's the way it came via bower and in GUI-1636, the file looks the same after we generated it: https://github.com/eucalyptus/eucaconsole/blob/GUI-1636/eucaconsole/static/css/euca-font-awesome.css

kamalgill commented on eucaconsole/static/css/pages/stack_wizard.css:
```diff
@@ -48,7 +48,9 @@
 .columns.value { padding-left: 1rem; }
 
-.keyvalsummary { line-height: 1.375rem; }
+.keyvalsummary { display: flex; justify-content: flex-start; line-height: 1.375rem; }
```
@dkavanagh: IE10 doesn't support the flex layout modes without vendor prefixes.  See http://caniuse.com/#feat=flexbox and http://stackoverflow.com/questions/18019450/css-flexbox-not-working-in-ie10

kamalgill commented on eucaconsole/static/sass/pages/stack_wizard.scss:
```diff
@@ -33,9 +33,18 @@
 }
 
 .keyvalsummary {
+    display: flex;
```
@dkavanagh: IE 10+ doesn't support un-prefixed `display: flex`

kamalgill commented on eucaconsole/templates/panels/top_nav.pt:
```diff
@@ -1,71 +1,61 @@
+    <li><a id="resource-menu-images" class="lnk-dashboard"
+           href="${request.route_path('images')}"><label i18n:translate="">Images</label></a></li>
+    <li class="has-dropdown">
+        <li><label id="resource-menu-instances" i18n:translate="">Instances</label></li>
```
@dkavanagh: An 'li' element with a child 'li' element isn't valid HTML.  Please revert to the 'a' element and find a way to fix the display via CSS.

kamalgill commented on eucaconsole/templates/stacks/stack_view.pt:
```diff
@@ -216,12 +206,15 @@
                                             </td>
                                             <td ng-show="item.url">
-                                                <a href="{{ item.url }}">{{ item.physical_id }}<a>
+                                                <a href="{{ item.url }}">{{ item.physical_id }}</a>
```
@dkavanagh: We need to use ng-href here instead of vanilla href since the URL is generated via an Angular expression.  See https://docs.angularjs.org/api/ng/directive/ngHref

kamalgill commented on eucaconsole/templates/stacks/stack_wizard.pt:
```diff
@@ -62,14 +62,14 @@
                                         <label i18n:translate="">Use sample template</label>
```
@dkavanagh: An input element may not contain child elements.  Please fix it here and elsewhere.  Note that PyCharm would highlight this and similar invalid HTML errors on the fly.

kamalgill commented on eucaconsole/templates/stacks/stack_wizard.pt:
```diff
@@ -62,14 +62,14 @@
+                                    <br ng-hide="inputtype=='sample'"/>
                                     <input type="radio" id="inputtype" name="inputtype" value="file" ng-model="inputtype">
                                         <label i18n:translate="">Upload template</label>
                                     </input>
```
@dkavanagh: Line 76 contains an input element with a separate closing tag, which isn't valid HTML.  Please modify it to use a self-closing singleton tag.

dkavanagh commented:
@kamalgill thanks for the feedback. I've fixed those things and some others. CI running now. I tested in IE11 with IE10 document and user agent set. Not sure if that's good enough though. The part of the page with flex looked good. I added the ms transitional css, so I hope that did the trick. We do have a lot of other stuff that looks bad in IE, mostly strings in fields though.

kamalgill commented:
@dkavanagh: Please back out the Selenium fixes for the fallout from GUI-1474 and open a distinct PR for those.

kamalgill commented on eucaconsole/templates/master_layout.pt:
```diff
@@ -21,7 +21,7 @@
     <script src="${request.static_path('eucaconsole:static/js/thirdparty/modernizr/custom.modernizr.js')}"></script>
     <script src="${request.static_path('eucaconsole:static/js/thirdparty/jquery/jquery.min.js')}"></script>
-    <script src="${request.static_path('eucaconsole:static/js/thirdparty/angular/angular.min.js')}"></script>
+    <script src="${request.static_path('eucaconsole:static/js/thirdparty/angular/angular.js')}"></script>
```
@dkavanagh: We should reference the minified file here

kamalgill commented on eucaconsole/templates/master_layout.pt:
```diff
@@ -78,6 +79,11 @@
 
+    <!-- Dropdown Menu: Left -->
+    <div id="nav-drop" class="f-dropdown">
+        ${panel('top_nav', off_canvas=False)}
```
@dkavanagh: Why is off_canvas False here?  I thought we're always using the off canvas nav now?

kamalgill commented on eucaconsole/static/sass/pages/stack_wizard.scss:
```diff
@@ -33,14 +33,35 @@
 }
 
 .keyvalsummary {
+    display: -ms-flexbox;
```
@dkavanagh: Sorry I didn't point this out earlier, but we'll also need to add vendor prefixes for Safari (using the newer flexbox spec, not the 2012 one).  See http://caniuse.com/#search=flex

dkavanagh commented:
Oops, the off-canvas=false was one more item I needed to remove.
I tested on safari 6.1.6 and it looks fine. I added in the older webkit tags called out here as well: https://css-tricks.com/using-flexbox/

euca-qa-gui-tester commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/196/
Test PASSed.

euca-qa-gui-tester commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/197/
Test FAILed.

dkavanagh commented:
hal, retest this please

kamalgill commented:
@dkavanagh: Closing w/o merging.   We need two things to happen before re-opening this PR... Please break out the selenium test fixes into a distinct branch (perhaps scoped in the GUI-1474 ticket), and please have the GUI-1489 branch pass UX review.

euca-qa-gui-tester commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/198/
Test PASSed.

euca-qa-gui-tester commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.qa1.eucalyptus-systems.com//job/user-console-PR-monitor/199/
Test PASSed.
